js/notebook/src/extension.js

```diff
@@ -158,17 +158,35 @@ define([
 
     // assign Beaker methods to window
     if (window) {
-      if (!window.beakerx) {
-        window.beakerx = {};
-      }
-
       var plotApiList = plotApi.list();
       var bkApp = bkCoreManager.getBkApp();
       var bkObject = bkApp.getBeakerObject();
 
-      _.extend(window.beakerx, plotApiList);
-      _.extend(window.beakerx, htmlOutput);
-      window.beakerx.prefs = bkObject.beakerObj.prefs;
+      var beakerxInstance = {}
+      _.extend(beakerxInstance, plotApiList);
+      _.extend(beakerxInstance, htmlOutput);
+      beakerxInstance.prefs = bkObject.beakerObj.prefs;
+
+      if (!window.beakerx) {
+        var handler = {
+          get: function (obj, prop) {
+            return prop in obj ? obj[prop] : undefined;
+          },
+
+          set: function (obj, prop, value) {
+            obj[prop] = value;
+            var comm = Jupyter.notebook.kernel.comm_manager.new_comm('beaker.autotranslation', null, null, null, utils.uuid());
```

Comment:
create this once and reuse it with each msg.
also, name the channel "beakerx.autotranslation".

Reply:
are all our channels named "beakerx.*"?

Reply:
nope, but I will change the names from `beaker.*` to `beakerx.*`